thavalon-webapp/src/components/gameComponents/interactions/missionManager.tsx

```diff
@@ -0,0 +1,153 @@
+            sendGameAction(GameActionType.Play, { card: card });
+            setHasPlayedCard(true);
+        }
+    }
+
+    // Create player cards here
+    const playerCards = props.playerList.map((playerName) => {
+        const selectedTypes = createSelectedPlayerTypesList(playerName, props.primarySelectedPlayers, props.secondarySelectedPlayers);
+        return <PlayerCard
+            key={playerName}
+            name={playerName}
+            vote={props.votes.get(playerName)}
+            tabbedOut={props.tabbedOutPlayers.has(playerName)}
+            selectedTypes={selectedTypes}
+            enabled={false} />
+    });
+
+    const playersOnMission = new Set(props.message.players);
+    return (
+        <>
+            {playerCards}
+            <div className="mission-manager">
+                {!hasPlayedCard && playersOnMission.has(props.me) &&
+                    <MissionCardButtons submitMissionCard={submitMissionCard} />}
+                {(hasPlayedCard || !playersOnMission.has(props.me)) &&
+                    // TODO: Support mission card progress bar here.
+                    <>Please wait for the mission cards to be played </>
+                }
+            </div>
+        </>
+    )
+
+}
+
+/**
+ * Component managing the card buttons a player interacts with
+ * @param props Required properties for the mission cards
+ */
+function MissionCardButtons(props: MissionCardProps): JSX.Element {
+    const buttons = new Array<JSX.Element>();
+    for (const card of Object.values(MissionCard)) {
+        buttons.push(
+            <button
+                key={card}
+                onClick={() => props.submitMissionCard(card)}>
+                {card}
+            </button>
+        );
+    }
+    return (
+        <div className="mission-card-buttons">
+            {buttons}
+        </div>
+    )
+}
+
+/**
+ * Component showing the results of a mission with Agravaine support
+ * @param props Required properties for the mission result modal
+ */
+export function MissionResultModal(props: MissionResultModalProps): JSX.Element {
+    if (props.message === undefined) {
+        return <></>;
+    }
+    const { successes, fails, reverses, questing_beasts, mission } = props.message;
+    return (
+        <ReactModal
+            isOpen={true}
+            onRequestClose={() => props.setOpen(false)}
+            contentLabel="Mission Result Modal"
+            className="Modal"
+            overlayClassName="Overlay"
+        >
+            <div className="modalContainer">
+                <h2 className="modalHeader">Misison {mission} {props.message.passed ? "Passed!" : "Failed!"}</h2>
+                <hr />
+                <ul>
+                    <li key={"successes"}>Successes: {successes}</li>
+                    <li key={"fails"}>Fails: {fails}</li>
+                    <li key={"reverses"}>Reverses: {reverses}</li>
+                    <li key={"questing_beasts"}>Questing Beasts {"<"}3: {questing_beasts}</li>
```

Comment:
{"<"} should probably be "\&lt;" (no quotes), I think that's more like HTML

Reply:
Fixed